### ISSUE TITLE
UISwitch binding doesn't work

### ISSUE THREAD

**Issue by PelleRavn** [MvvmCross/MvvmCross#2462]
### Steps to reproduce :scroll:

1. Create view with a UISwitch on it 
2. Implement binding 🔗
```c#
var set = this.CreateBindingSet<FirstView, FirstViewModel>();
set.Bind(Switch).For(s => s.On).To(vm => vm.TermsAccepted);
set.Apply(); 
```
3. Run and show View 🏃‍♂️

### Expected behavior :thinking:
Binding would work when changing state for the switch, without any error.

### Actual behavior :bug:
Binding error in application output:
`MvxBind:Error: 10.50 Problem creating target binding for UISwitch - exception System.MissingMethodException: Constructor on type 'MvvmCross.Binding.iOS.Target.MvxUISwitchOnTargetBinding' not found.`

### Possible solution 👀
Might be related to the recent commit: https://github.com/MvvmCross/MvvmCross/commit/8073c40862e79a279ba38767f1cb13a109adca0f
Fix could be same as with #2456 

### Configuration :wrench:

**Version:** 5.6.2

**Platform:** 
- [X] :iphone: iOS
- [ ] :robot: Android
- [ ] :checkered_flag: WPF
- [ ] :earth_americas: UWP
- [ ] :apple: MacOS
- [ ] :tv: tvOS
- [ ] :monkey: Xamarin.Forms

**Comment by modplug** [comment]
I got this error as well

**Comment by willsb** [comment]
This should only happen when the linker is on; it's stripping the code away. Easily fixed by adding the following snippet to your LinkerPleaseInclude.cs:

```csharp
public void Include(MvxUISwitchOnTargetBinding binding)
{
    binding = new MvxUISwitchOnTargetBinding(null, null);
}
```

MvvmCross contains a lot of its core target bindings in the LinkerPleaseInclude file, but if we add all of the bindings there, it defeats the purpose of the linker. Those should be added manually (I have done it myself in the past, as seen [here](https://github.com/toggl/mobileapp/blob/develop/Toggl.Daneel/LinkerPleaseInclude.cs)

**Comment by PelleRavn** [comment]
@willsb That should not be a solution to this, at all. 
First of all, this have never been an issue before 5.6.x, so if I have to start adding every single target binding I use, in my LinkerPleaseInclude, it would be completely madness, and nothing in my app would work anymore (as I only use the default one).

Second, this error happens even with the linker off. It's either something with the change in the constructor (as the error message says `Constructor on type 'MvvmCross.Binding.iOS.Target.MvxUISwitchOnTargetBinding' not found.`), or the solution to the issue I referenced in the post.

**Comment by willsb** [comment]
It is a solution to this, yes, and I'm getting the exact same error on MvvmCross 5.2.

**Comment by tougher** [comment]
Just got this after upgrading to 5.6.x

**Comment by Cheesebaron** [comment]
@PelleRavn why isn't this a solution?

This is a balance between preserving everything and resulting in a big fat assembly or having the developer choose what they want to keep. There are tools to help you debug and get insights into what the linker strips out.

I think the better solution would be to provide commented out sections in the LinkerPleaseInclude file we provide, where devs can remove comments to prevent stuff from being stripped out that comes from MvvmCross.

Also saying "linking is disabled" then deploying to a device, means nothing, since it enables the linker unconditionally in that case.

**Comment by PelleRavn** [comment]
Why it isn't a solution: Because it doesn't change anything. If I add it to my LinkerPleaseInclude, it makes no difference - the error still happens.

I did some digging, and the problem is, as stated earlier, the constructor itself.
The second parameter that was removed (`PropertyInfo targetPropertyInfo`) is trying to be parsed into the constructor when created at runtime. https://github.com/MvvmCross/MvvmCross/commit/8073c40862e79a279ba38767f1cb13a109adca0f#diff-7026059ee96c3c00122e1b0b4e6d66f9L21

So if the constructor is changed from the current one:
```C#
public MvxUISwitchOnTargetBinding(UISwitch target)
     : base(target)
{
}
```

back to the old constructor before the commit:
```C#
public MvxUISwitchOnTargetBinding(UISwitch target, PropertyInfo targetPropertyInfo)
     : base(target)
{
}
```
everything works as expected.

I wanted to make a PR with this, but my old fork is misbehaving, so I'm unable to do that at the moment.

**Comment by Cheesebaron** [comment]
OK, Thanks for elaborating. I will have a look at this issue.

**Comment by Cheesebaron** [comment]
This is fixed in #2467. You can work around it by:

adding these two usings in your `Setup.cs` file:

```csharp
using MvvmCross.Binding.Bindings.Target.Construction;
using MvvmCross.Binding.iOS.Target;
```

overriding `FillTargetFactories`:

```csharp
protected override void FillTargetFactories(IMvxTargetBindingFactoryRegistry registry)
{
    base.FillTargetFactories(registry);

    registry.RegisterCustomBindingFactory<UISwitch>(
        "On",
        uiSwitch => new MvxUISwitchOnTargetBinding(uiSwitch)
    );
}
```

Important here is to call base first, so we override the existing registration for `UISwitch`.
